3rdparty/fastcv/src/fastcv_hal_imgproc.cpp

```diff
@@ -743,4 +743,253 @@ int fastcv_hal_warpPerspective(
         CV_HAL_RETURN_NOT_IMPLEMENTED(cv::format("Src type:%s is not supported", cv::typeToString(src_type).c_str()));
 
     CV_HAL_RETURN(status, hal_warpPerspective);
+}
+
+class FcvPyrLoop_Invoker : public cv::ParallelLoopBody
+{
+public:
+
+    FcvPyrLoop_Invoker(cv::Mat src_, int width_, int height_, cv::Mat dst_, int bdr_, int knl_, int stripeHeight_, int nStripes_) :
+        cv::ParallelLoopBody(), src(src_), width(width_), height(height_), dst(dst_), bdr(bdr_), knl(knl_), stripeHeight(stripeHeight_), nStripes(nStripes_)
+    {
+    }
+
+    virtual void operator()(const cv::Range& range) const CV_OVERRIDE
+    {
+        int height_ = stripeHeight * (range.end - range.start);
+        int width_  = width;
+        cv::Mat src_;
+        int n = knl/2;
+
+        if(range.end == nStripes)
+            height_ += (height - range.end * stripeHeight);
+
+        src_ = cv::Mat(height_ + 2*n, width_ + 2*n, CV_8U);
+
+        if(range.start == 0 && range.end == nStripes)
+            cv::copyMakeBorder(src(cv::Rect(0, 0, width, height)), src_, n, n, n, n, bdr);
+        else if(range.start == 0)
+            cv::copyMakeBorder(src(cv::Rect(0, 0, width_, height_ + n)), src_, n, 0, n, n, bdr);
+        else if(range.end == nStripes)
+            cv::copyMakeBorder(src(cv::Rect(0, range.start * stripeHeight - n, width_, height_ + n)), src_, 0, n, n, n, bdr);
+        else
+            cv::copyMakeBorder(src(cv::Rect(0, range.start * stripeHeight - n, width_, height_ + 2*n)), src_, 0, 0, n, n, bdr);
+
+        int dstHeight_, dstWidth_, origDstHeight_, origDstWidth_;
+        dstHeight_ = (height_ + 2*n + 1)/2;
+        dstWidth_ = (width_ + 2*n + 1)/2;
+        origDstHeight_ = (height_ + 1)/2;
+        origDstWidth_ = (width_ + 1)/2;
+
+        cv::Mat dst_padded = cv::Mat(dstHeight_, dstWidth_, CV_8U);
+
+        fcvPyramidLevel_v2 *framePyr = new fcvPyramidLevel_v2[2];
```

Comment:
I propose to use local variable on stack. new is redundant here. Also it's not deleted afterwards.